docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+    tree.Set(id, post)
+}
+```
+
+In this example, `GetPost` is a function that retrieves a post from the
+`avl.Tree` using an ID. It only loads the post with the specified ID, without
+loading any other posts.
+
+In the future, we plan to add internal "map" support that will be as efficient
+as an `avl.Tree` but with a more idiomatic API. For now, consider storing things
+in slices when you know the structure will stay small, and consider using
+`avl.Tree` each time you can make direct access.
+
+You can also create SQL-like indexes by having multiple `avl.Tree` instances for
+different fields. For example, you can have an `avl.Tree` for ID to *post, then
+an `avl.Tree` for Tags, etc. Then, you can create reader methods that will just
+retrieve what you need, similar to SQL indexes.
+
+By using `avl.Tree` and other efficient data structures, you can optimize your
+Gno code for performance and cost-effectiveness, making your applications more
+scalable and efficient.
+
+TODO: multi-indices example
+
+### Construct "safe" objects
+
+A safe object in Gno is an object that is designed to be tamper-proof and
+secure. It's created with the intent of preventing unauthorized access and
+modifications. This follows the same principle of making a package an API, but
+for a Go object that can be directly referenced by other realms.
+
+The goal is to create an object which, once instantiated, can be linked and its
+pointer can be "stored" by other realms without issue, because it protects its
+usage completely.
+
+```go
+type MySafeStruct {
+    counter nb
+    admin std.Address
+}
+
+func NewSafeStruct() *MySafeStruct {
+    caller := std.GetOrigCaller()
+    return &MySafeStruct{
+        counter: 0,
+        admin: caller,
+    }
+}
+
+func (s *MySafeStruct) Counter() int { return s.counter }
+func (s *MySafeStruct) Inc() {
+    caller := std.GetOrigCaller()
+    if caller != s.admin {
+        panic("permission denied")
+    }
+    s.counter++
+}
+```
+
+Then, you can register this object in another or several other realms so other
+realms can access the object, but still following your own rules.
+
+```go 
+import "gno.land/r/otherrealm"
+
+func init() {
+    mySafeObj := NewSafeStruct()
+    otherrealm.Register(mySafeObject)
+}
+
+// then, otherrealm can call the public functions but won't be the "owner" of
+// the object.
+```
+
+### Choosing between native tokens and GRC20 tokens
+
+In Gno, you've got two main choices for tokens: Native or GRC20. Each has its
+own pros and cons, and the best fit depends on your needs.
+
+#### Native tokens
+
+Native tokens are managed by the banker module, separate from Gnovm. They're
+simple, strict, and secure. You can create, transfer, and check balances with an
+RPC call, no Gnovm needed.
+
+For example, if you're creating a coin for cross-chain transfers, native tokens
+are your best bet. They're IBC-ready and their strict rules offer top-notch
+security.
+
+TODO: native banker example.
```

Comment:
Leftover?

Reply:
Resolved as part of https://github.com/gnolang/gno/pull/1000#discussion_r1434195360